point_viewer_grpc/src/service.rs

```diff
@@ -48,6 +50,26 @@ enum OctreeQuery {
     FullPointcloud,
 }
 
+fn send_fail_stream<T>(ctx: &RpcContext, sink: ServerStreamingSink<T>, error: &Error) {
+    let f = sink
+        .fail(RpcStatus::new(
+            RpcStatusCode::Internal,
+            Some(error.to_string()),
+        ))
+        .map_err(move |err| eprintln!("Failed to reply: {:?}", err));
+    ctx.spawn(f);
+}
+
+fn send_fail<T>(ctx: &RpcContext, sink: UnarySink<T>, error: &Error) {
```

Comment:
These two functions have exactly the same bodies - is it possible to have just one, a little more generic, function here?

Reply:
I tried and thought about it, but the two sinks [ServerStreamingSink](https://docs.rs/grpcio/0.4.4/grpcio/struct.ServerStreamingSink.html) and [UnarySink](https://docs.rs/grpcio/0.4.4/grpcio/struct.UnarySink.html) have no common trait. They have different return types for the `fail` method, so I couldn't find a way to simplify it... Do you know one?

Reply:
I can't think of any way to simplify it...